dspace-api/src/main/java/org/dspace/discovery/configuration/DiscoveryConfigurationService.java

```diff
@@ -22,9 +32,18 @@
  */
 public class DiscoveryConfigurationService {
 
+    private static final Logger log = LogManager.getLogger();
+
     private Map<String, DiscoveryConfiguration> map;
     private Map<Integer, List<String>> toIgnoreMetadataFields = new HashMap<>();
 
+    /**
+     * Discovery configurations, cached by Community/Collection UUID. When a  Community or Collection does not have its
+     * own configuration, we take the one of the first parent that does.
+     * This cache ensures we do not have to go up the hierarchy every time.
+     */
+    private final Map<UUID, DiscoveryConfiguration> comColToDiscoveryConfigurationMap = new HashMap<>();
```

Comment:
@YanaDePauw ok we can keep it simple and don't use a thread bound map but in this case we need to pick a Thread-safe implementation like the `ConcurrentHashMap` otherwise we could get issues il multiple users make changes to this map. Please note that it is not completely true that community and collection cannot be moved as we have a script to do that but this is not a very frequent scenario that would require in any case a reindex, so I'm ok with that. @tdonohue would be better to add a message in the `org.dspace.administer.CommunityFiliator` output to remind about the need to restart tomcat and reindex the content?

Reply:
@abollini : While I'm not against adding a message to CommunityFiliator, I think we can also simply just add a note to the documentation here: https://wiki.lyrasis.org/display/DSDOC7x/Managing+Community+Hierarchy

In fact, I'm going to add that note immediately, simply because I suspect reindexing is already necessary (it's just undocumented).  The only new requirement we are adding in this PR is that you likely also need to restart Tomcat in order to reset any cached community/collection information.